### DISCUSSION TITLE
Documentation Overhaul

### DISCUSSION THREAD

**Discussion by damienwebdev** [faker-js/faker#87] [Ideas]
In service of making this repo easier to consume (both by users and maintainers) we need to improve the documentation. 

Here, I note important issues, discussions, etc that are relevant:

## Issues

- [x] https://github.com/faker-js/faker/issues/22
    - [x] https://github.com/faker-js/faker/issues/46
    - [x] https://github.com/faker-js/faker/issues/49
- [x] https://github.com/faker-js/faker/issues/18
- [x] https://github.com/faker-js/faker/issues/33
- [x] https://github.com/faker-js/faker/issues/73

## Ideal End State for Docs

Generally speaking, I think most of these docs should be a part of this codebase, not a separate app. The chief concern here is that the further the docs are from the code, the more likely they are to become stale.

- [x] Autogenerated API docs based upon doc blocks on methods and properties
    - [x] Depends on Typescript Migration
    - [x] https://github.com/faker-js/faker/pull/82
- [x] General guides for usage
    - [x] Should exist in the `docs` folder
        - [x]  Review what is already in the docs folder
- [ ] Written tutorials
    - [ ] General walkthrough of the how to setup and install the code
        - [ ] In a simple browser project
        - [ ] In a NodeJS project
        - [ ] In a Typescript project
- [ ] A general playground for messing with Faker

**Comment by ericjeker** [comment]
This PR is for TypeDoc: https://github.com/faker-js/faker/pull/82

If we want to use TSDoc instead we should close it. Depends on your preferences. I think Jess already started using TypeDoc but it's just about generating a JSON output so no big deal.

**Reply by damienwebdev** [reply]
@ericjeker I've modified the doc above, I have no preference either way.
